### PR TITLE
fix(typescript): TS2322: Type 'null' is not assignable to type 'strin…

### PR DESCRIPTION
**Pull request by Caballerog**
…g | void'

Added the type Html (String | undefined) which allow assign null to string when
you use the flags: --strictNullChecks (TS 2.2)

### CODE REVIEW

File: src/quill-editor.component.ts [hunk 1]
```diff
@@ -178,11 +179,11 @@ export class QuillEditorComponent implements AfterViewInit, ControlValueAccessor
...
     let err: {
-      minLengthError?: {given: number, minLength: number};
-      maxLengthError?: {given: number, maxLength: number};
-      requiredError?: {empty: boolean}
+      minLengthError?: { given: number, minLength: number };
+      maxLengthError?: { given: number, maxLength: number };
+      requiredError?: { empty: boolean }
     } = {},
-    valid = true;
+      valid = true;
```
Comment (KillerCodeMonkey) [inline src/quill-editor.component.ts]: why that space? if you want to indent this --> indent the whole block even the err object

File: src/quill-editor.component.ts [hunk 2]
```diff
@@ -126,11 +127,11 @@ export class QuillEditorComponent implements AfterViewInit, ControlValueAccessor
 
     // update model if text changes
     this.quillEditor.on('text-change', (delta: any, oldDelta: any, source: string) => {
-      let html = this.editorElem.children[0].innerHTML;
+      let html: Html = this.editorElem.children[0].innerHTML;
```
Comment (KillerCodeMonkey) [inline src/quill-editor.component.ts]: why you want to type this at all? If so then type it directly here as `html: string? = .....` optional string.

HTML is a false leading name --> HTMLString or something like this.

File: src/quill-editor.component.ts [hunk 3]
```diff
@@ -116,7 +117,7 @@ export class QuillEditorComponent implements AfterViewInit, ControlValueAccessor
         range: range,
         oldRange: oldRange,
         source: source,
-        bounds: this.bounds || document.body
+        bounds: this.bounds ||  document.body
```
Comment (KillerCodeMonkey) [inline src/quill-editor.component.ts]: duplicated space

File: src/quill-editor.component.ts [hunk 4]
```diff
@@ -74,14 +75,14 @@ export class QuillEditorComponent implements AfterViewInit, ControlValueAccessor
   @Input() minLength: number;
   @Input() required: boolean;
   @Input() formats: string[];
-  @Input() bounds: HTMLElement | string;
+  @Input() bounds: HTMLElement |  string;
```
Comment (KillerCodeMonkey) [inline src/quill-editor.component.ts]: duplicated string

File: src/quill-editor.component.ts [hunk 5]
```diff
@@ -20,6 +20,7 @@ import {
 
 import * as Quill from 'quill';
 
+type Html = string | null;
```
Comment (KillerCodeMonkey) [inline src/quill-editor.component.ts]: see the comment line 130

File: src/quill-editor.component.ts (hunk 4 above)
Comment (Caballerog) [inline src/quill-editor.component.ts]: Sorry, this change is from VScode (my linter in autosave). There is not relevant to the commit.

File: src/quill-editor.component.ts (hunk 3 above)
Comment (Caballerog) [inline src/quill-editor.component.ts]: Sorry, this change is from VScode (my linter in autosave). There is not relevant to the commit.

File: src/quill-editor.component.ts (hunk 1 above)
Comment (Caballerog) [inline src/quill-editor.component.ts]: Sorry, this change is from VScode (my linter in autosave). There is not relevant to the commit.

File: src/quill-editor.component.ts (hunk 2 above)
Comment (Caballerog) [inline src/quill-editor.component.ts]: Quick background, if you are not using --strictNullChecks both null and undefined are in the domain of all types. so T | undefined | null is equivalent to T. if you are using --strictNullChecks however, that is not the case. T does not include null or undefined.(https://github.com/Microsoft/TypeScript/issues/8322).

So, the following code is not a valid in TS: 
     ```
 if (html === '<p><br></p>') {
        html = null;
      }
```
So, I thought use the range (String and null) created a new type "html" (Maybe the name is not the more suitable. Futhermore, I also tried your posible solution:

`let html: string? = ...`

But It is not a valid TS for my version.

In the new angular version 4.1 strictNullCheck flag come to be the default behaviour in AOT (Which is disable in the version 4.0.2).

Comment (KillerCodeMonkey) [inline src/quill-editor.component.ts]: okay it was only a shot in the dark but `let html: string | null = ....` should work at all? in other cases rename the type `Html` to maybe `contentHTMLString` or something like that.

Change these things --> make a rebase and i will merge it and create a new version.

thanks

Comment (Caballerog) [inline src/quill-editor.component.ts]: You're right.
It's not necessary to create a new type, using the following definition is sufficient:
`let html: (string | null) = ...`
Thanks.

**Pull request closed without merging**